### PR TITLE
Add RoslynTools.BuildTasks

### PR DESCRIPTION
**Pull request by tmat**

### CODE REVIEW

**Comment by jaredpar** [comment]
What is this task going to be used for?

**Comment by tmat** [comment]
Various tasks needed by the toolset. There are few currently - download file (for dependency uptake), resx generation - x-plat replacement of ResGen previously implemented by .csx file in the toolset, URL escaping for source link tooling.

**Comment by tmat** [comment]
PRs using these tasks will follow.

File: src/BuildTasks/RoslynTools.BuildTasks.csproj [hunk 1]
```diff
@@ -0,0 +1,20 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <TargetFramework>netstandard1.3</TargetFramework>
+    <IsPackable>true</IsPackable>
+    <Description>RoslynTools build tasks</Description>
+    <PackageTags>RoslynTools msbuild task</PackageTags>
+    <IsTool>true</IsTool>
+
+    <!-- Using an explicit nuspec file since NuGet Pack target currently doesn't support including dependencies in tools packages -->
+    <IsPackable>true</IsPackable>
```
Comment (jaredpar) [inline src/BuildTasks/RoslynTools.BuildTasks.csproj]: The IsPackable property si duplicated in this file

Comment (tmat) [reply src/BuildTasks/RoslynTools.BuildTasks.csproj]: Thanks, will clean up.

**Pull request merged**